### PR TITLE
TS styles

### PR DESCRIPTION
**Pull request by chrisnojima**
This uses styles from the `@types of RN/react`. In a lot of places a raw string type is now an explicit list of strings, therefore the objects we use need to be const so as to not get widened
practically speaking that means

old:
```
Styles.styleSheetCreate({...}) // etc
```

new: (closure for dark mode)
```
Styles.styleSheetCreate(() => ({...} as const ))
```

https://github.com/keybase/client/pull/19273/files?w=1 is your friend

### CODE REVIEW

File: shared/tracker2/bio/index.tsx [hunk 1]
```diff
@@ -180,42 +180,45 @@ const Bio = (p: Props) => (
...
-  floatingContainer: Styles.platformStyles({
-    common: {
-      backgroundColor: Styles.globalColors.purple,
-    },
-    isElectron: {
-      maxWidth: 200,
-    },
-  }),
-  fullName: Styles.platformStyles({
-    isElectron: {wordBreak: 'break-any'},
```
Comment (chrisnojima) [inline shared/tracker2/bio/index.tsx]: incorrect!

File: shared/styles/css.d.ts [hunk 2]
```diff
@@ -1,516 +1,35 @@
-export type Color = null | string
-type Dimension = string | number
+import {CSSProperties} from 'react'
```
Comment (chrisnojima) [inline shared/styles/css.d.ts]: the 'meat' of the changes in this PR

Comment (chrisnojima) [inline shared/styles/css.d.ts]: pull in styles from react and RN

File: shared/styles/css.d.ts [hunk 3]
```diff
@@ -1,516 +1,35 @@
...
-  textDecorationColor?: Color
-  writingDirection?: 'auto' | 'ltr' | 'rtl'
-  tintColor?: Color
-  overlayColor?: string
+type _StylesCrossPlatform = {
+  [k in keyof _StylesDesktop]: k extends keyof _StylesCrossPlatformOverride // use override
+    ? _StylesCrossPlatformOverride[k] // or if its shared between desktop and mobile choose one which extends the other
+    : k extends keyof _StylesMobile
+    ? _StylesMobile[k] extends _StylesDesktop[k]
+      ? _StylesMobile[k]
```
Comment (chrisnojima) [inline shared/styles/css.d.ts]: we prefer mobile to desktop as its more constrained. the css styles from react mostly are of the form
` 'left' | 'right' | undefined | string ` that last `| string` makes that type just string which is very annoying

File: shared/styles/index.types.d.ts [hunk 4]
```diff
@@ -1,103 +0,0 @@
-// Styles from our designers
```
Comment (chrisnojima) [inline shared/styles/index.types.d.ts]: not used

File: shared/styles/status-bar.d.ts [hunk 5]
```diff
@@ -1,3 +0,0 @@
-import {EmitterSubscription} from 'react-native'
```
Comment (chrisnojima) [inline shared/styles/status-bar.d.ts]: not used

File: shared/people/follow-notification/index.tsx [hunk 6]
```diff
@@ -101,32 +101,35 @@ export const MultiFollowNotification = (props: Props) => {
...
+        },
+        isMobile: {
+          ...Styles.globalStyles.flexBoxRow,
+          flexWrap: 'wrap',
+          height: 32,
+          overflow: 'hidden',
+          width: '100%',
+        },
+      }),
+      usernames: Styles.platformStyles({isElectron: {whiteSpace: 'normal'} as const}),
```
Comment (buoyad) [inline shared/people/follow-notification/index.tsx]: Impressively the only storyshot change in here 😎

**Pull request merged**